includes/class-query-integration.php

```diff
@@ -282,12 +280,26 @@ public function posts_clauses( $pieces, $query ) {
 		}
 
 		$pieces['fields']         = apply_filters( 'wpbdp_query_fields', $pieces['fields'] );
-		$pieces['custom_orderby'] = apply_filters( 'wpbdp_query_orderby', ( $is_sticky_query ? 'wpbdp_is_sticky DESC' : '' ) );
+		$pieces['custom_orderby'] = apply_filters( 'wpbdp_query_orderby', ( $sticky_ids_str ? "FIELD(wp_posts.ID,{$sticky_ids_str}) DESC" : '' ) );
 		$pieces['orderby']        = ( $pieces['custom_orderby'] ? $pieces['custom_orderby'] . ', ' : '' ) . $pieces['orderby'];
-
 		return $pieces;
 	}
 
+	/**
+	 * Select all sticky listing ids and return thim in string comma seperated.
+	 *
+	 * @return string listing ids
+	 */
+	private function get_sticky_listing_ids() {
+		global $wpdb;
+		$results = $wpdb->get_results( "SELECT listing_id FROM {$wpdb->prefix}wpbdp_listings  WHERE is_sticky=1" );
```

Comment:
Let's run the query through our check_cache() functions too.

Reply:
Thanks for the update!

The group here should be for the table getting searched. This helps us keep track of when the cache should be cleared.

`'group' => 'wpbdp_listings',`

Reply:
I made this change and merged in my changes from the other pr.